enos/tests/utils/test_extra.py

```diff
@@ -332,6 +339,156 @@ def test_load_provider_config_with_missing_keys(self):
                 provider_config,
                 default_provider_config)
 
+class TestSSH(unittest.TestCase):
+    longMessage = True
+    hosts = [Host('1.2.3.4')]
+    env = {'resultdir': 'foo/bar', 'inventory': 'foo/bar'}
+
+    def test_wait_ssh_succeed(self):
+        with mock.patch('enos.utils.extra.run_ansible',
+                        new_callable=mock.Mock()) as m:
+            m.return_value=None
+            self.assertIsNone(wait_ssh(self.env, interval=0))
+
+    def test_wait_ssh_eventually_succeed(self):
+        with mock.patch('enos.utils.extra.run_ansible',
+                        new_callable=mock.Mock()) as m:
+            effects = [EnosUnreachableHostsError(self.hosts)
+                       for i in range(1, 10)]
+            effects.append(None)
+            m.side_effect = effects
+            self.assertIsNone(wait_ssh(self.env, retries=10, interval=0))
+
+    def test_wait_ssh_fails(self):
+        with self.assertRaisesRegexp(Exception, 'Maximum retries reached'),\
+             mock.patch('enos.utils.extra.run_ansible',
+                        new_callable=mock.Mock()) as m:
+            m.side_effect=EnosUnreachableHostsError(self.hosts)
+            wait_ssh(self.env, interval=0)
+
+
+class TestPathLoading(unittest.TestCase):
+    longMessage = True
+
+    def setUp(self):
+        self.sourcedir = const.ENOS_PATH
+        self.workdir = os.path.realpath(tempfile.mkdtemp())
+
+    def tearDown(self):
+        shutil.rmtree(self.workdir)
+
+    def assertPathEqual(self, p1, p2, msg=None):
+        self.assertEqual(os.path.normpath(p1),
+                         os.path.normpath(p2),
+                         msg)
+
+    def test_seek_inventory_path(self):
```

Comment:
One idea : 
Make use of ddt here to factorize the two tests : 
https://ddt.readthedocs.io/en/latest/example.html

This could be something : 
```
@ddt.data(('/abs/path/to/inventory.sample', 'inventories/inventory.sample'),
                   ('/abs/path/to/workload/', 'workload/'))
@unpack
def test_seek_inventory_path(self, abspath, relpath):
```